sdk/storage/src/datalake/file_client.cpp

```diff
@@ -0,0 +1,287 @@
+    protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
+    protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
+    protocolLayerOptions.IfModifiedSince = options.AccessConditions.IfModifiedSince;
+    protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
+    protocolLayerOptions.SourceIfMatch = options.SourceAccessConditions.IfMatch;
+    protocolLayerOptions.SourceIfNoneMatch = options.SourceAccessConditions.IfNoneMatch;
+    protocolLayerOptions.SourceIfModifiedSince = options.SourceAccessConditions.IfModifiedSince;
+    protocolLayerOptions.SourceIfUnmodifiedSince = options.SourceAccessConditions.IfUnmodifiedSince;
+    protocolLayerOptions.RenameSource = "/" + m_dfsUri.GetPath();
+    auto result = DataLakeRestClient::Path::Create(
+        destinationDfsUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
+    // At this point, there is not more exception thrown, meaning the rename is successful.
+    m_dfsUri = std::move(destinationDfsUri);
+    m_blobUri = Details::GetBlobUriFromDfsUri(m_dfsUri);
+    auto ret = FileRenameResponse();
+    ret.Date = std::move(result.Date);
+    ret.ETag = std::move(result.ETag);
+    ret.LastModified = std::move(result.LastModified);
+    ret.RequestId = std::move(result.RequestId);
+    ret.Version = std::move(result.Version);
+    return ret;
+  }
+
+  FileDeleteResponse FileClient::Delete(const FileDeleteOptions& options) const
+  {
+    DataLakeRestClient::Path::DeleteOptions protocolLayerOptions;
+    // TODO: Add null check here when Nullable<T> is supported
+    protocolLayerOptions.LeaseIdOptional = options.AccessConditions.LeaseId;
+    protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
+    protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
+    protocolLayerOptions.IfModifiedSince = options.AccessConditions.IfModifiedSince;
+    protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
+    auto result = DataLakeRestClient::Path::Delete(
+        m_dfsUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
+    auto ret = FileDeleteResponse();
+    ret.Date = std::move(result.Date);
+    ret.RequestId = std::move(result.RequestId);
+    ret.Version = std::move(result.Version);
+    return ret;
+  }
+
+  ReadPathResponse FileClient::Read(const FileReadOptions& options) const
+  {
+    DataLakeRestClient::Path::ReadOptions protocolLayerOptions;
+    if (options.Offset.HasValue())
+    {
+      auto rangeString = std::string("bytes=" + std::to_string(options.Offset.GetValue()) + "-");
+      if (options.Length.HasValue())
+      {
+        rangeString += std::to_string(options.Offset.GetValue() + options.Length.GetValue() - 1);
+      }
+      protocolLayerOptions.Range = std::move(rangeString);
+    }
+
+    protocolLayerOptions.XMsRangeGetContentMd5 = options.RangeGetContentMD5;
```

Comment:
```suggestion
    protocolLayerOptions.RangeGetContentMd5 = options.RangeGetContentMD5;
```

Reply:
Protocol layer code that transparent to customer.